### PR TITLE
Layout updating

### PR DESCRIPTION
**Pull request by vketrys**
- [x] Layout updating
- [x] Fixing buttons 
- [x] Fixing gaps
- [x] Creating language switcher
- [x] Added a Ticker component on Main Page 

Снимок экрана от 2022-09-12 13-40-44

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,8 @@
     "react-redux": "7.2.6",
     "react-router-dom": "6.2.2",
     "react-scripts": "5.0.0",
+    "react-switch": "^7.0.0",
```
Comment (kozlova-anzhelika) [inline package.json:24]: remove ^

Comment (kozlova-anzhelika) [inline package.json:24]: remove ^

Comment (vketrys) [reply package.json:24]: yeah, sure

File: src/components/switcher/Switcher.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        onChange={handleChange}
+        checked={languageRu}
+        offColor="#fff"
+        onColor="#fff"
+        checkedIcon={
+          <img
+            src={IconEN}
+            height={25}
+            alt="EN"
+            style={{ margin: '2px 0 0 2px' }}
```
Comment (nemets-i) [inline src/components/switcher/Switcher.tsx:33]: doesn't use inline styles, please

Comment (vketrys) [reply src/components/switcher/Switcher.tsx:33]: ok, i'll fix that

File: src/components/switcher/Switcher.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    } else i18n.changeLanguage('ru');
+  };
+
+  return (
+    <div>
+      <ReactSwitch
+        onChange={handleChange}
+        checked={languageRu}
+        offColor="#fff"
+        onColor="#fff"
```
Comment (nemets-i) [inline src/components/switcher/Switcher.tsx:27]: Move colors to constants, please

File: src/pages/main/Main.tsx [hunk 4]
```diff
@@ -32,6 +38,13 @@ function MainPage(): JSX.Element {
           onClick={(): void => navigate(routerPaths.episode)}
         />
       </div>
+      <div className={style.bottom}>
+        {tickerPhrases.map((el, index) => (
+          <Ticker offset={(index + 1) * 50} speed={25} mode="await">
```
Comment (nemets-i) [inline src/pages/main/Main.tsx:43]: What do these magic numbers mean? Move to constants, please.

File: src/components/switcher/Switcher.module.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.image {
+  margin: 2px 0 0 2px;
```
Comment (slamser56) [inline src/components/switcher/Switcher.module.scss:2]: indent(0.5)?

File: src/pages/main/Main.module.scss [hunk 6]
```diff
@@ -20,5 +20,19 @@
 
   h1 {
     font-size: $Label;
+    height: 250px;
+    width: 650px;
+  }
+
+  .bottom {
+    margin-top: 60px;
```
Comment (slamser56) [inline src/pages/main/Main.module.scss:28]: use indent

File: src/config/locales/en/translation.json [hunk 7]
```diff
@@ -27,7 +27,10 @@
     "page": "Explorer",
     "characterButton": "Characters",
     "locationButton": "Locations",
-    "episodeButton": "Episodes"
+    "episodeButton": "Episodes",
+    "tickerPhrase1": "Let's go!",
+    "tickerPhrase2": "In and out",
+    "tickerPhrase3": "20 minutes adventure..."
```
Comment (slamser56) [inline src/config/locales/en/translation.json:33]: Move to separate object in translation object

File: src/components/switcher/Switcher.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+function Switcher(): JSX.Element {
+  const { i18n } = useTranslation();
+
+  const [languageRu, setLanguageRu] = useState(false);
+
+  const handleChange = (): void => {
+    setLanguageRu(!languageRu);
+    if (languageRu) {
+      i18n.changeLanguage('en');
```
Comment (slamser56) [inline src/components/switcher/Switcher.tsx:20]: Use enum or const to language keys

File: src/components/switcher/Switcher.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import IconEN from 'img/EN.png';
+import IconRU from 'img/RU.png';
+
+import colors from 'variables/colors';
+import style from './Switcher.module.scss';
+
+function Switcher(): JSX.Element {
+  const { i18n } = useTranslation();
+
+  const [languageRu, setLanguageRu] = useState(false);
```
Comment (slamser56) [inline src/components/switcher/Switcher.tsx:15]: Are you really sure it is a good solution?

File: src/pages/main/Main.tsx [hunk 10]
```diff
@@ -32,6 +39,17 @@ function MainPage(): JSX.Element {
...
       </div>
+      <div className={style.bottom}>
+        {tickerPhrases.map((el, index) => (
+          <Ticker
+            offset={(index + 1) * tickerVars.baseOffset}
+            speed={tickerVars.speed}
+            mode="await"
+          >
+            {(): JSX.Element => <p>{el}</p>}
+          </Ticker>
```
Comment (slamser56) [inline src/pages/main/Main.tsx:50]: I guess it should be memoized component

File: src/pages/main/Main.tsx [hunk 11]
```diff
@@ -1,22 +1,29 @@
...
 
 function MainPage(): JSX.Element {
   const { t } = useTranslation();
   const navigate = useNavigate();
 
+  const tickerPhrases = [
+    t('main.tickerPhrase1'),
+    t('main.tickerPhrase2'),
+    t('main.tickerPhrase3'),
+  ];
```
Comment (slamser56) [inline src/pages/main/Main.tsx:21]: Move outside component

File: src/components/switcher/Switcher.tsx (hunk 9 above)
Comment (vketrys) [reply src/components/switcher/Switcher.tsx:15]: are you about naming ?

Comment (slamser56) [reply src/components/switcher/Switcher.tsx:15]: I asked you about your solution

Comment (vketrys) [reply src/components/switcher/Switcher.tsx:15]: i'll fix that

Comment (slamser56) [reply src/components/switcher/Switcher.tsx:15]: Here you are using isn't flexible solution instead you can use the key for any language

File: src/config/locales/ru/translation.json [hunk 12]
```diff
@@ -26,7 +26,10 @@
     "page": "Энциклопедия",
     "characterButton": "Персонажи",
     "locationButton": "Локации",
-    "episodeButton": "Эпизоды"
+    "episodeButton": "Эпизоды",
+    "tickerPhrase1": "Давай!",
+    "tickerPhrase2": "Пошли и вышли",
+    "tickerPhrase3": "Приключение на 20 минут..."
```
Comment (slamser56) [inline src/config/locales/ru/translation.json:32]: remove unused translate

File: src/pages/main/Main.tsx [hunk 13]
```diff
@@ -32,6 +37,17 @@ function MainPage(): JSX.Element {
...
+      <div className={style.bottom}>
+        {tickerPhrases.map((el, index) => (
+          <MemoTicker
+            offset={(index + 1) * tickerVars.baseOffset}
+            speed={tickerVars.speed}
+            mode="await"
+          >
+            {(): JSX.Element => <p>{el}</p>}
+          </MemoTicker>
+        ))}
```
Comment (slamser56) [inline src/pages/main/Main.tsx:49]: move to separate memoized component

File: src/pages/main/Main.tsx [hunk 14]
```diff
@@ -1,22 +1,27 @@
...
 
 import style from './Main.module.scss';
 
 function MainPage(): JSX.Element {
   const { t } = useTranslation();
   const navigate = useNavigate();
 
+  const tickerPhrases: string[] = t('ticker', { returnObjects: true });
+
+  const MemoTicker = React.memo(Ticker);
```
Comment (slamser56) [inline src/pages/main/Main.tsx:19]: Does it matter?

File: src/components/switcher/Switcher.tsx [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+        setLanguage(languagesKeys.ru);
+        break;
+      case 'ru':
+        setLanguage(languagesKeys.en);
+        break;
+      default:
+        setLanguage(languagesKeys.ru);
+        break;
+    }
+    i18n.changeLanguage(language);
```
Comment (slamser56) [inline src/components/switcher/Switcher.tsx:33]: Does your switcher work?

File: src/config/locales/ru/translation.json (hunk 12 above)
Comment (vketrys) [reply src/config/locales/ru/translation.json:32]: yeah sure

File: src/pages/main/Main.tsx (hunk 14 above)
Comment (vketrys) [reply src/pages/main/Main.tsx:19]: it doesn't ??
okay, I should export separate component like this

File: src/components/switcher/Switcher.tsx (hunk 15 above)
Comment (vketrys) [reply src/components/switcher/Switcher.tsx:33]: yes, it's work

File: src/components/switcher/Switcher.tsx [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+import IconRU from 'img/RU.png';
+
+import languagesKeys from 'variables/languages';
+import colors from 'variables/colors';
+import style from './Switcher.module.scss';
+
+function Switcher(): JSX.Element {
+  const { i18n } = useTranslation();
+
+  const [switcher, setSwitcher] = useState(false);
```
Comment (slamser56) [inline src/components/switcher/Switcher.tsx:16]: isSwitched?

Comment (vketrys) [reply src/components/switcher/Switcher.tsx:16]: yeah, that's will be better naming

**Review by kozlova-anzhelika — APPROVED** [review]

**Review by i-severinchik — APPROVED** [review]

**Review by slamser56 — APPROVED** [review]

**Pull request merged**